File: backend/app/schemas/workout_insights.py
from pydantic import BaseModel


class PlateauEntry(BaseModel):
    exercise_id: int
    exercise_name: str
    sessions_without_progress: int
    current_weight_kg: float


class DeloadSuggestion(BaseModel):
    consecutive_weeks_trained: int
    suggested: bool
    message: str


class WorkoutInsightsResponse(BaseModel):
    plateaus: list[PlateauEntry]
    deload: DeloadSuggestion

File: mobile/src/api/workoutInsights.ts
import { api } from "./client";

export type PlateauEntry = {
  exercise_id: number;
  exercise_name: string;
  sessions_without_progress: number;
  current_weight_kg: number;
};

export type DeloadSuggestion = {
  consecutive_weeks_trained: number;
  suggested: boolean;
  message: string;
};

export async function getWorkoutInsights(): Promise<{
  plateaus: PlateauEntry[];
  deload: DeloadSuggestion;
}> {
  const { data } = await api.get("/workout-insights");
  return data;
}
